matrix_client/client.py

```diff
@@ -574,7 +574,15 @@ def upload(self, content, content_type):
             )
 
     def _mkroom(self, room_id):
-        self.rooms[room_id] = Room(self, room_id)
+        room = Room(self, room_id)
+        if self.encryption:
+            try:
+                self.api.get_encryption_algorithm(room_id)
```

Comment:
This technically needs to check whether the room was ever encrypted and whether the algorithm was valid, right?

Reply:
Absolutely, this was really bad.